File: app/src/main/java/com/android/asa/ui/splash/SplashActivity.java
package com.android.asa.ui.splash;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import com.android.asa.R;
import com.android.asa.ui.signin.SignInActivity;

public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);


        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                navigateToLogin();
            }
        }, 1500);
    }

    private void navigateToLogin() {
        Intent intent = new Intent(this, SignInActivity.class);
        startActivity(intent);
        finish();
    }
}

File: app/src/main/java/com/android/asa/ui/splash/SplashActivity.kt
package com.android.asa.ui.splash

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import com.android.asa.MainActivity
import com.android.asa.R
import com.android.asa.ui.auth.AuthenticationActivity
import com.android.asa.ui.staff_advisor.StaffHomeActivity
import com.asa.data.sharedPreference.SharedPreferenceReader
import com.google.firebase.auth.FirebaseAuth
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.delay
import javax.inject.Inject

@AndroidEntryPoint
class SplashActivity : AppCompatActivity() {

    @Inject
    lateinit var fireBaseAuth: FirebaseAuth

    @Inject
    lateinit var preferenceReader: SharedPreferenceReader

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        lifecycleScope.launchWhenCreated {
            delay(1500)
            val user = preferenceReader.getUserData()
            if (fireBaseAuth.currentUser != null && user != null) {
                if (user.userType == 0) navigateToMainActivity()
                else navigateToStaffModule()
            } else {
                navigateToLogin()
            }
        }
    }

    fun navigateToStaffModule() {
        startActivity(StaffHomeActivity.intent(this))
        finish()
    }

    private fun navigateToLogin() {
        val intent = Intent(this, AuthenticationActivity::class.java)
        startActivity(intent)
        finish()
    }

    private fun navigateToMainActivity() {
        val intent = Intent(this, MainActivity::class.java)
        startActivity(intent)
        finish()
    }
}